crates/aptos/src/common/utils.rs

```diff
@@ -154,11 +155,23 @@ pub fn check_if_file_exists(file: &Path, prompt_options: PromptOptions) -> CliTy
 }
 
 pub fn prompt_yes_with_override(prompt: &str, prompt_options: PromptOptions) -> CliTypedResult<()> {
-    if prompt_options.assume_no || (!prompt_options.assume_yes && !prompt_yes(prompt)) {
-        Err(CliError::AbortedError)
-    } else {
-        Ok(())
+    if prompt_options.assume_no {
+        return Err(CliError::AbortedError);
+    }
+
+    if !prompt_options.assume_yes {
+        match GlobalConfig::load()?.get_default_prompt_response() {
```

Comment:
A couple of points here:

First, GlobalConfig only reads the config at the global folder (e.g. `~/.aptos`), we want to read the correct config based on scope. So if there is an `.aptos` dir in the current dir, use that. You want something like this instead:
```
let config = CliConfig::load(ConfigSearchMode::CurrentDirAndParents)?;
```

Second, we shouldn't load the config here, it should be passed in, since in some cases we might already have it. That way we avoid unnecessary double reads of the config.

Reply:
First - As `default_prompt_response` is saved in global config file, I think using `GlobalConfig::load` rather makes sense. `CliConfig::load` loads the profiles only. Otherwise, it should be stored in the profiles as needed.
Second - I agree with your principle. `GlobalConfig::load` is used to get `.aptos` folder path mostly to load profiles, save them, and check the config existence and called more than once to load and then save in a line. Furthermore, it's not designed to pass the global config from the root to the leaf, which means it should be redesigned and/or needed much updates relatively.

@banool, thoughts?

Reply:
First: Okay I may be misunderstanding the code, I'd have to dive deeper. Though for now I'll invoke @gregnazario to comment since he knows the CLI better than anyone 😛 

Second: Same thing, what do you think @gregnazario? I see instances where we have a config already, it seems redundant to load it again. Perhaps the config object should be smarter and act as a passthrough cache.

Reply:
Sorry for the delay, been very very busy.

I would put it in the global config like he does here, and I think it's the right choice.

This code maybe should read a slightly different way.  This will let you override via CLI flags default behavior even if you sent it, allow you to set a default which overtakes so you don't need to send flags, and allow for default prompting otherwise.

This pseudocode below I think lets it be very simple and straightforward for someone to read afterwards and tell what's going on.


```
// Handle CLI flags
If assume_no:
 return abort
else if assume_yes:
 return Ok

// Then handle default or prompt behavior
bool is_yes = if let Some(response) = default_response:
  response
else:
  prompt and get response

if is_yes:
  return Ok
else:
  return abort
```

Reply:
thanks for the reply. I just did some changes. could you please review again?